2333/2/libs/top_words_analyzer.rb

```diff
@@ -0,0 +1,29 @@
+require_relative 'handler'
+require_relative './analyzer_printer.rb'
+require_relative './names_searcher.rb'
+
+# VersusBattle text analyzer
+# :reek:ControlParameter
+class TopWordsAnalyzer
+  def initialize(rappers, selected_name, take_value)
+    @rappers = rappers
+    @name = selected_name
+    @take_value = take_value || 30
+  end
+
+  def top_words
+    AnalyzerPrinter.print_rappers_names(names, @name) unless names_include_name?
+    counter = texts_wo_stopwords.each_with_object(Hash.new(0)) { |word, hash| hash[word.downcase] += 1 }
```

Comment:
You create a temporary `counter` variable. You can avoid this. See how: https://github.com/rubizza-survival-camp/summer-2018/pull/251/files#diff-cdbb98d77e7a36751f9837686ed61ba1R18.